### ISSUE TITLE
The sync command is popping a stash every time

### ISSUE THREAD

**Issue by rhys-vdw** [ForkIssues/TrackerWin#1765]
After hitting "sync" my colleague was immediately getting a host of conflicts that did not exist. Turns out they were coming from a random stash that already existed.

We were testing with branches that were already at parity with origin/master.

We tried deleting all the stashes (`git stash clear`), but then sync fails with this error:

```
[Internal]: System.IO.FileNotFoundException: Could not find file 'W:\enter-the-chronosphere.git\logs\refs\stash'.
File name: 'W:\enter-the-chronosphere.git\logs\refs\stash'
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.StreamReader..ctor(String path, Encoding encoding, Boolean detectEncodingFromByteOrderMarks, Int32 bufferSize, Boolean checkHost)
   at System.IO.File.InternalReadAllText(String path, Encoding encoding, Boolean checkHost)
   at System.IO.File.ReadAllText(String path)
   at Fork.Git.Commands.LeanBranching.LeanBranching.Step0Stash(GitModule gitModule, JobMonitor monitor)
```

So now the sync button is unusable.

OS: Windows
Version: 1.80.0

**Comment by DanPristupov** [comment]
You need to remove the `.git/fork/sync` directory manually.

Please let me know if you see that problem again.

**Comment by DanPristupov** [comment]
> `W:\enter-the-chronosphere.git`

What configuration do you use? Where is the working directory located?

**Comment by rhys-vdw** [comment]
Sorry, this was on my colleague's computer so I'll need to check with her where the working directory is. I doubt it's `W:` (not sure if that would even be possible).

The issue persists so I can look into it further.

**Comment by rhys-vdw** [comment]
Ah, I should also mention that if we create a new stash, sync will try to pop it every time. So clearly something is busted here.

Is there some way to modify the sync command so it simply does `git pull --rebase origin/master`? That's all we need.

**Comment by rhys-vdw** [comment]
FYI the working directory was `W:\enter-the-chronosphere`, ~~so that error message suggests a path issue (should be `'W:\enter-the-chronosphere\.git\logs\refs\stash`).~~ When we went to delete the `.git/fork/sync` it was not there. Sync now works (after upgrading to today's release, or perhaps manually using rebase pull on master last week). So problem solved itself.

EDIT: Correction below

**Comment by rhys-vdw** [comment]
Hey @DanPristupov, so this is still ongoing.

>`W:\enter-the-chronosphere.git`
>What configuration do you use? Where is the working directory located?

This was my mistake. We copied this message through work Discord, which interpreted the \ in `\.git` as an escape character. This is the correct error message after clearing stash:

```
[Internal]: System.IO.FileNotFoundException: Could not find file 'W:\enter-the-chronosphere\.git\logs\refs\stash'.
File name: 'W:\enter-the-chronosphere\.git\logs\refs\stash'
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.StreamReader..ctor(String path, Encoding encoding, Boolean detectEncodingFromByteOrderMarks, Int32 bufferSize, Boolean checkHost)
   at System.IO.File.InternalReadAllText(String path, Encoding encoding, Boolean checkHost)
   at Fork.Git.Commands.LeanBranching.LeanBranching.Step0Stash(GitModule gitModule, JobMonitor monitor) 
```

Before clearing stash we just get huge numbers of asset changes unrelated to the current commit. This is when syncing a branch that is already up-to-date with `origin/master`. There is clearly something very wrong here, but I don't know what "sync" does to cause this.

Basically it is just not possible for my colleague to use "sync" at present.

**Comment by rhys-vdw** [comment]
We are STILL getting this issue in 2.3.1.0, this time on a new team member's machine.

After syncing their branch (with no changes, staged or unstaged) a random stash is applied _after_ successfully rebasing and pushing.

```
# Stash uncommitted changes:

$ git stash save --message "Fork sync autostash 12/12/2024 5:06:25 PM"

No local changes to save

# Synchronize 'eff-5628-radgunner-purifier-animation--without-animation-bug' with 'origin/eff-5628-radgunner-purifier-animation--without-animation-bug':

$ git -c core.commentChar=^ rebase refs/remotes/origin/eff-5628-radgunner-purifier-animation--without-animation-bug

Current branch eff-5628-radgunner-purifier-animation--without-animation-bug is up to date.

# Synchronize 'eff-5628-radgunner-purifier-animation--without-animation-bug' with 'refs/remotes/origin/master':

$ git -c core.commentChar=^ rebase refs/remotes/origin/master

Current branch eff-5628-radgunner-purifier-animation--without-animation-bug is up to date.

# Move 'origin/eff-5628-radgunner-purifier-animation--without-animation-bug' to 'refs/heads/eff-5628-radgunner-purifier-animation--without-animation-bug':

$ git push origin refs/heads/eff-5628-radgunner-purifier-animation--without-animation-bug:refs/heads/eff-5628-radgunner-purifier-animation--without-animation-bug --force-with-lease --verbose

Pushing to https://github.com/EffortStar/enter-the-chronosphere.git
To https://github.com/EffortStar/enter-the-chronosphere.git
 = [up to date]            eff-5628-radgunner-purifier-animation--without-animation-bug -> eff-5628-radgunner-purifier-animation--without-animation-bug
updating local tracking ref 'refs/remotes/origin/eff-5628-radgunner-purifier-animation--without-animation-bug'
Everything up-to-date
# Apply stashed changes:

$ git stash pop stash@{0}

Auto-merging Assets/_Game/Art/Enemies/WarMonkeys/Animation/Purifier/Purifier@Idle.fbx
CONFLICT (content): Merge conflict in Assets/_Game/Art/Enemies/WarMonkeys/Animation/Purifier/Purifier@Idle.fbx
Auto-merging Assets/_Game/Art/Enemies/WarMonkeys/Animation/Purifier/Purifier@Idle.fbx.meta
CONFLICT (content): Merge conflict in Assets/_Game/Art/Enemies/WarMonkeys/Animation/Purifier/Purifier@Idle.fbx.meta
CONFLICT (rename/delete): Assets/_Game/Content/Bosses/BulletHellComponents/FireBurstAtPlayer.meta renamed to Assets/_Game/Art/Environment/OuterWall.meta in Updated upstream, but deleted in Stashed changes.
CONFLICT (modify/delete): Assets/_Game/Art/Environment/OuterWall.meta deleted in Stashed changes and modified in Updated upstream.  Version Updated upstream of Assets/_Game/Art/Environment/OuterWall.meta left in tree.
CONFLICT (rename/delete): Assets/_Game/Content/Bosses/BulletHellComponents/StoppedProjectile.meta renamed to Assets/_Game/Art/VisualProfiles.meta in Updated upstream, but deleted in Stashed changes.
CONFLICT (modify/delete): Assets/_Game/Art/VisualProfiles.meta deleted in Stashed changes and modified in Updated upstream.  Version Updated upstream of Assets/_Game/Art/VisualProfiles.meta left in tree.
On branch eff-5628-radgunner-purifier-animation--without-animation-bug
Your branch is up to date with 'origin/eff-5628-radgunner-purifier-animation--without-animation-bug'.

Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	modified:   Assets/_Game/Materials/SquiggleShadowUnstable.mat

Unmerged paths:
  (use "git restore --staged <file>..." to unstage)
  (use "git add/rm <file>..." as appropriate to mark resolution)
	both modified:   Assets/_Game/Art/Enemies/WarMonkeys/Animation/Purifier/Purifier@Idle.fbx
	both modified:   Assets/_Game/Art/Enemies/WarMonkeys/Animation/Purifier/Purifier@Idle.fbx.meta
	deleted by them: Assets/_Game/Art/Environment/OuterWall.meta
	deleted by them: Assets/_Game/Art/VisualProfiles.meta

The stash entry is kept in case you need it again.
Encountered 1 file that should have been a pointer, but wasn't:
	Assets/_Game/Art/Enemies/WarMonkeys/Animation/Purifier/Purifier@Idle.fbx
```

NOTE: No stash is made, yet a random stash is popped.

**Comment by DanPristupov** [comment]
We will investigate this.

**Comment by rhys-vdw** [comment]
> We will investigate this.

Thank you! Its terrible for our non-technical team members.

While you're at it, I noticed that the "sync" button incorrectly says `rebase on 'master'` instead of `rebase on 'origin/master'` to perform this operation.

![image](https://github.com/user-attachments/assets/77849dbb-4f4c-4e13-a3f5-dc447f4bf187)

However you can see that it does actually rebase on `origin/master`:

```
$ git -c core.commentChar=^ rebase refs/remotes/origin/master
```

This caused a lot of confusion for me, because sometimes it does say `rebase on 'origin/master'` (I think when `master` itself is checked out).

For a while I was instructing our team to check out master, pull, and then go back to their branch and sync due to this misleading name.

**Comment by DanPristupov** [comment]
The original problem is fixed in 2.4.
